### PR TITLE
woodpecker-*: 1.0.1 -> 1.0.2

### PR DESCRIPTION
**Pull request by ambroisie**
The release has not yet been made on GitHub, so no links.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/continuous-integration/woodpecker/common.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
 { lib, fetchFromGitHub }:
 let
-  version = "1.0.1";
-  srcHash = "sha256-uwKLD3fW/em6UMkkyrWxAo7T//Hkzj6WjIp5qJVtBuc=";
+  version = "1.0.2";
+  srcHash = "sha256-OeOKR9lTGXU2qumyXq3s/CI56IX4XiI/ZTRHNoY6MEI=";
   vendorHash = "sha256-NYWJorVeRxbQTiirHK8gqpDddn2RsKsNWwDNdcONVQA=";
```
Comment (adamcstephens) [inline pkgs/development/tools/continuous-integration/woodpecker/common.nix:5]: go.mod changed in this release, so vendorHash needs to be updated.

```
┃          specified: sha256-NYWJorVeRxbQTiirHK8gqpDddn2RsKsNWwDNdcONVQA=
┃             got:    sha256-+2wQKNyCb9xtB1TeE1/oSMEvKoXVX9ARZxsNqE2rfrg=
```

**Review by adamcstephens — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/continuous-integration/woodpecker/common.nix (hunk 1 above)
Comment (ambroisie) [reply pkgs/development/tools/continuous-integration/woodpecker/common.nix:5]: Not sure how that was missed, as I used the update script 🤔.

**Comment by 6543** [comment]
CVE-2023-40034 :)

File: pkgs/development/tools/continuous-integration/woodpecker/common.nix (hunk 1 above)
Comment (ambroisie) [reply pkgs/development/tools/continuous-integration/woodpecker/common.nix:5]: That's already the case ;-)

Comment (adamcstephens) [reply pkgs/development/tools/continuous-integration/woodpecker/common.nix:5]: Oops, I grep'd the wrong location. 🙈

**Review by adamcstephens — APPROVED** [review]
Result of `nixpkgs-review pr 249559` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>woodpecker-agent</li>
    <li>woodpecker-cli</li>
    <li>woodpecker-server</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 249559` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>woodpecker-agent</li>
    <li>woodpecker-cli</li>
    <li>woodpecker-server</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 249559` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>woodpecker-agent</li>
    <li>woodpecker-cli</li>
    <li>woodpecker-server</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 249559` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>woodpecker-agent</li>
    <li>woodpecker-cli</li>
    <li>woodpecker-server</li>
  </ul>
</details>

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Pull request merged**